tuf-spec.md

```diff
@@ -1479,13 +1494,55 @@ it in the next step.
   2. Otherwise, recursively search the list of delegations in
      order of appearance.
 
-    1. If the current delegation is a terminating delegation,
+    1. If the current delegation is a multi-role delegation,
+       recursively visit each delegated role, and check that each has signed
+       exactly the
+       same non-custom metadata (i.e., length and hashes) about the target (or
+       the lack of any such metadata). Otherwise, abort the update cycle, and
+       report the failure.
+
+    2. Let DELEGATEE denote
+       the current target role DELEGATOR is delegating to.
+
+    3. **Download the DELEGATEE targets metadata file**, up to either
+       the number of bytes specified in the snapshot metadata file, or some Z
+       number of bytes. The value for Z is set by the authors of the application
+       using TUF. For example, Z may be tens of kilobytes. IF DELEGATEE cannot be
+       found, end the search and report the target cannot be found.  If
+       consistent snapshots are not used (see Section
+       [[#consistent-snapshots]]), then the filename used
+       to download the targets metadata file is of the fixed form FILENAME.EXT
+       (e.g., delegated_role.json).  Otherwise, the filename is of the form
+       VERSION_NUMBER.FILENAME.EXT (e.g., 42.delegated_role.json), where
+       VERSION_NUMBER is the version number of the DELEGATEE metadata file listed
+       in the snapshot metadata file.  In either case, the client MUST write the
+       file to non-volatile storage as FILENAME.EXT.
+
+    4. **Check against snapshot metadata.** The hashes (if any), and
+       version number of the new DELEGATEE metadata file MUST match the trusted
+       snapshot metadata, if any.  This is done, in part, to prevent a mix-and-match
```

Comment:
the "if any" has confused me, shouldn't we always have a trusted snapshot metadata at this point? as we download snapshot metadata before targets.